provider/ec2/environ_test.go

```diff
@@ -407,3 +409,37 @@ func (*Suite) TestGetValidSubnetZoneMapIntersectionSelectsCorrectIndex(c *gc.C)
 	c.Assert(err, jc.ErrorIsNil)
 	c.Check(subnetZones, gc.DeepEquals, allSubnetZones[1])
 }
+
+func (*Suite) TestGatherNilAZ(c *gc.C) {
+	az := gatherAvailabilityZones(nil)
+	c.Assert(az, gc.HasLen, 0)
+}
+
+func (*Suite) TestGatherEmptyAZ(c *gc.C) {
+	instances := []instances.Instance{}
+	az := gatherAvailabilityZones(instances)
+	c.Assert(az, gc.HasLen, 0)
+}
+
+func (*Suite) TestGatherAZ(c *gc.C) {
+	instances := []instances.Instance{
+		&amzInstance{
+			Instance: &amzec2.Instance{
+				AvailZone: "aaa",
+			},
+		},
+		&sdkInstance{
+			i: &ec2.Instance{
+				Placement: &ec2.Placement{
+					AvailabilityZone: ptrString("bbb"),
+				},
+			},
+		},
+	}
+	az := gatherAvailabilityZones(instances)
+	c.Assert(az, gc.DeepEquals, []string{"aaa", "bbb"})
+}
+
+func ptrString(s string) *string {
```

Comment:
This is fine, though note that there's also the SDK function `aws.String` for this.